### DISCUSSION TITLE
Add more unit options for gaseous SensorDeviceClasses

### DISCUSSION THREAD

**Discussion by kamiyo** [home-assistant/architecture#1082] [Entity Models]
Currently, some of the gaseous sensors only allow certain units that do not always correspond to various standards. For example, for certain pollutants in the US, the EPA prefers to use ppm or ppb for some of them:
[https://www.airnow.gov/sites/default/files/2020-05/aqi-technical-assistance-document-sept2018.pdf](https://www.airnow.gov/sites/default/files/2020-05/aqi-technical-assistance-document-sept2018.pdf)

Ozone (ppm)
CO (ppm)
SO2 (ppb)
NO2 (ppb)

Right now, only CO allows ppm.

Inversely, for the EU, it seems that all of them are usually μg per cubic meter, except CO, which is mg per cubic meter: [https://environment.ec.europa.eu/topics/air/air-quality/eu-air-quality-standards_en](https://environment.ec.europa.eu/topics/air/air-quality/eu-air-quality-standards_en).

I propose we add ppm option to ozone, ppb options to SO2 and NO2, and μg per cubic meter to CO sensors.

I don't know if putting multiple units requires automatic conversion between them. Part of the problem with converting between parts-per-volume and mass-per-volume for gasses is that assumptions must be made about temperature and pressure. We could just use 20 deg C and 1 atm (which I think almost all agencies use, though some in the US use 25 deg C), but we'd still have to know the molecular weight of each compound (which we could hardcode, but still have to pass in as a parameter to the conversion).

I know this may seem sort of inane to think about, but I think it'd be nice to support by default these different units in the included core sensor classes. Thanks for considering.

**Comment by joostlek** [comment]
What integrations in core could use this right now?

**Reply by kamiyo** [reply]
The one that prompted me to look this up was AirNow. The web API returns Ozone in ppm instead of micrograms. As a result the AirNow integration just uses a custom sensor type. But since the ozone class exists, it'd be nice to use that.
